### PR TITLE
cannon: supplement the example cannon command

### PR DESCRIPTION
**Pull request by DenseDenise**
Re-open https://github.com/ethereum-optimism/optimism/pull/11762
And fixed comment referred to the review comment by @smartcontracts

### CODE REVIEW

File: cannon/README.md [hunk 1]
```diff
@@ -48,7 +49,7 @@ make cannon
     --input ./state.bin.gz \
     -- \
     ../op-program/bin/op-program \
-    --network op-mainnet \
+    --network <Predefined_Network> \
```
Comment (smartcontracts) [inline cannon/README.md]:
```suggestion
    --network <network name> \
```

File: cannon/README.md [hunk 2]
```diff
@@ -39,6 +39,7 @@ make cannon
 # Note:
 #  - The L2 RPC is an archive L2 node on OP MAINNET.
 #  - The L1 RPC is a non-archive RPC, also change `--l1.rpckind` to reflect the correct L1 RPC type.
+#  - The network flag is only suitable for specific networks(https://github.com/ethereum-optimism/superchain-registry/blob/main/chainList.json). If you are running on the devnet, please use '--l2.genesis' with the L2 devnet genesis file.
```
Comment (smartcontracts) [inline cannon/README.md]:
```suggestion
#  - The network flag is only suitable for specific networks(https://github.com/ethereum-optimism/superchain-registry/blob/main/chainList.json). If you are running on the devnet, please use '--l2.genesis' to supply a path to the L2 devnet genesis file.
```

I believe this is correct

Comment (DenseDenise) [reply cannon/README.md]: Yeah!

**Review by smartcontracts — APPROVED** [review]

**Comment by smartcontracts** [comment]
thanks!

**Pull request merged**